### ISSUE TITLE
Molecule.add_conformer doesn't accept openff-unit units

### ISSUE THREAD

**Issue by richardjgowers** [openforcefield/openff-toolkit#1249]
**Describe the bug**

Openff seems to only accept openmm units.

**To Reproduce**

```
In [1]: from openff.toolkit.topology import Molecule

In [2]: from openff.units import unit

In [3]: import numpy as np

In [4]: m = Molecule.from_smiles('CCO')

In [5]: coords = np.random.random((len(m.atoms), 3))

In [6]: m.add_conformer(coords * unit.angstrom)

```
**Output**

`Exception: Coordinates passed to Molecule._add_conformer without units. Ensure that coordinates are of type openmm.units.Quantity`

**Additional context**

I tried looking through issues and I think there's a move away from openmm units in progress(?), but I'm not sure if that's actually happening on the topology refactor branch.  Sorry if this has been fixed there.  I think it wouldn't be too nasty to have some sort of decorator around functions with typed inputs which converted the openff-unit units to openmm units.  I think an alternative solution might be to have openff-units quack like an openmm unit when required, but maybe that's more of an openff-unit repo issue than a toolkit issue.

**Comment by mattwthompson** [comment]
Are you using version 0.10.x?

**Comment by richardjgowers** [comment]
@mattwthompson 0.10.3 of tk and 0.1.5 of units

**Comment by mattwthompson** [comment]
Okay yeah, sorry, that behavior isn't landing until version 0.11.0 and unfortunately we're only backporting critical bugfixes to 0.10.x.  [This](https://github.com/openforcefield/openff-toolkit/blob/d5c465e21c1fc900c156dc697afc3be3562f4718/openff/toolkit/tests/test_molecule.py#L2493) is the test that's supposed to check that behavior and somehow I neglected to cover that - my oversight.  The relevant source code looks fine and runs your reproduction without error, so I guess that's why I forgot to add it in the test:

>  I think it wouldn't be too nasty to have some sort of decorator around functions with typed inputs which converted the openff-unit units to openmm units.

Hmm, I hadn't considered this. It sure seems like it would be an improvement over the branching logic I've used this far ...

> I think an alternative solution might be to have openff-units quack like an openmm unit when required, but maybe that's more of an openff-unit repo issue than a toolkit issue.

I agree, although I've had a hard time pulling that off for reasons that escape memory (https://github.com/openforcefield/openff-units/pull/9)
